src/logic/hooks/useEstimateGasLimit.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react'
+
+const DEFAULT_GAS_LIMIT = '0'
+
+export const useEstimateGasLimit = (
+  estimationFn: () => Promise<number>,
+  isExecution: boolean,
+  txData: string,
+  manualGasLimit: string | undefined,
+): string | undefined => {
+  const [gasLimit, setGasLimit] = useState<string | undefined>(manualGasLimit)
+
+  useEffect(() => {
+    let isCurrent = true
+
+    if (manualGasLimit) {
+      setGasLimit(manualGasLimit)
+      return
+    }
+
+    if (!isExecution || !txData) {
+      return
+    }
+
+    const estimateGasLimit = async () => {
+      try {
+        const estimation = await estimationFn()
+        isCurrent && setGasLimit(estimation.toString())
+      } catch (e) {
+        console.log(e)
```

Comment:
Please use logError instead.

Reply:
Also added another test to make sure we get the default gas estimation if the function throws.